aptible/resource_environment_test.go

```diff
@@ -0,0 +1,138 @@
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
+		Steps:             testSteps,
+	})
+}
+
+func testAccCheckEnvironmentDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*aptible.Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aptible_environment" {
+			continue
+		}
+
+		envID, err := strconv.Atoi(rs.Primary.Attributes["env_id"])
+		if err != nil {
+			return err
+		}
+
+		metricDrain, err := client.GetEnvironment(int64(envID))
+		log.Println("Deleted? ", metricDrain.Deleted)
+		if !metricDrain.Deleted {
+			return fmt.Errorf("environment %v not removed", envID)
+		}
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func TestAccResourceEnvironment_basic(t *testing.T) {
+	rHandle := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAptibleEnvironment(rHandle),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aptible_environment.test", "handle", rHandle),
+					resource.TestCheckResourceAttr("aptible_environment.test", "org_id", testOrganizationId),
+					resource.TestCheckResourceAttr("aptible_environment.test", "stack_id", strconv.Itoa(testStackId)),
+				),
+			}, {
+				ResourceName:            "aptible_environment.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"org_id", "stack_id"},
+			},
+		},
+	})
+}
+
+func TestAccResourceEnvironment_update(t *testing.T) {
+	rHandle := acctest.RandString(10)
+	rUpdatedHandle := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAptibleEnvironment(rHandle),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aptible_environment.test", "handle", rHandle),
+					resource.TestCheckResourceAttr("aptible_environment.test", "org_id", testOrganizationId),
+					resource.TestCheckResourceAttr("aptible_environment.test", "stack_id", strconv.Itoa(testStackId)),
+				),
+			}, {
+				ResourceName:      "aptible_environment.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			}, {
+				Config: testAccAptibleEnvironment(rUpdatedHandle),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aptible_environment.test", "handle", rUpdatedHandle),
+					resource.TestCheckResourceAttr("aptible_environment.test", "org_id", testOrganizationId),
+					resource.TestCheckResourceAttr("aptible_environment.test", "stack_id", strconv.Itoa(testStackId)),
+				),
+			},
```

Comment:
More of a note to myself. I'd like to determine if there's a good way to test for an update as opposed to a recreate since the resulting attributes are the same.

Reply:
Moving this to the Terraform FAQ doc.